azurerm/data_source_storage_account_sas.go

```diff
@@ -0,0 +1,390 @@
+							Required: true,
+						},
+
+						"queue": {
+							Type:     schema.TypeBool,
+							Required: true,
+						},
+
+						"table": {
+							Type:     schema.TypeBool,
+							Required: true,
+						},
+
+						"file": {
+							Type:     schema.TypeBool,
+							Required: true,
+						},
+					},
+				},
+			},
+
+			// Always in UTC and must be ISO-8601 format
+			"start": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			// Always in UTC and must be ISO-8601 format
+			"expiry": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"permissions": {
+				Type:     schema.TypeList,
+				Required: true,
+				ForceNew: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"read": {
+							Type:     schema.TypeBool,
+							Required: true,
+						},
+
+						"write": {
+							Type:     schema.TypeBool,
+							Required: true,
+						},
+
+						"delete": {
+							Type:     schema.TypeBool,
+							Required: true,
+						},
+
+						"list": {
+							Type:     schema.TypeBool,
+							Required: true,
+						},
+
+						"add": {
+							Type:     schema.TypeBool,
+							Required: true,
+						},
+
+						"create": {
+							Type:     schema.TypeBool,
+							Required: true,
+						},
+
+						"update": {
+							Type:     schema.TypeBool,
+							Required: true,
+						},
+
+						"process": {
+							Type:     schema.TypeBool,
+							Required: true,
+						},
+					},
+				},
+			},
+
+			"sas": {
+				Type:      schema.TypeString,
+				Computed:  true,
+				Sensitive: true,
+			},
+		},
+	}
+
+}
+
+func dataSourceArmStorageAccountSasRead(d *schema.ResourceData, _ interface{}) error {
+
+	connString := d.Get("connection_string").(string)
+	httpsOnly := d.Get("https_only").(bool)
+	resourceTypesIface := d.Get("resource_types").([]interface{})
+	servicesIface := d.Get("services").([]interface{})
+	start := d.Get("start").(string)
+	expiry := d.Get("expiry").(string)
+	permissionsIface := d.Get("permissions").([]interface{})
+
+	resourceTypes := buildResourceTypesString(resourceTypesIface[0].(map[string]interface{}))
+	services := buildServicesString(servicesIface[0].(map[string]interface{}))
+	permissions := buildPermissionsString(permissionsIface[0].(map[string]interface{}))
+
+	_, svcErr := validateArmStorageAccountSasResourceTypes(resourceTypes, "")
+	if svcErr != nil {
+		return svcErr[0]
+	}
+
+	// Parse the connection string
+	kvp, err := parseAzureStorageAccountConnectionString(connString)
+	if err != nil {
+		return err
+	}
+
+	// Create the string to sign with the key...
+
+	// Details on how to do this are here:
+	// https://docs.microsoft.com/en-us/rest/api/storageservices/Constructing-an-Account-SAS
+	accountName := kvp[connStringAccountNameKey]
+	accountKey := kvp[connStringAccountKeyKey]
+	var signedProtocol = "https,http"
+	if httpsOnly {
+		signedProtocol = "https"
+	}
+	signedIp := ""
```

Comment:
should we also expose this value?

Reply:
I debated this. The format is a little odd of the field and it's not all that functionally useful.

I say we omit it for now and if anyone asks for this added feature we can add it later.